os-autoinst-obs-auto-submit

```diff
@@ -129,7 +129,7 @@ generate_os-autoinst-distri-opensuse-deps_changelog() {
         echo "-------------------------------------------------------------------"
         echo "$(LANG=c date -u) - Dominik Heidler <dheidler@suse.de>"
         echo
-        diff -u "$package"-factory.spec "$dir"/"$package"/_service:obs_scm:"$package".spec \
+        { diff -u "$package"-factory.spec "$dir"/"$package"/_service:obs_scm:"$package".spec || :; } \
```

Comment:
*sigh* can we extract that into a function to prevent the confusion about the duplicate code? Like

```bash

spec_has_changes_in_requires() {
    local dir="$1"
    local package="$2"
    { diff -u "$package"-factory.spec "$dir"/"$package"/_service:obs_scm:"$package".spec || :; } | grep "^[+-]Requires"
}
…
-    { diff -u "$package"-factory.spec "$dir"/"$package"/_service:obs_scm:"$package".spec || :; } | grep "^[+-]Requires"
+    spec_has_changes_in_requires $dir $package
```

Reply:
I didn't even notice it is indeed diffing the very same files. Yes that's a good candidate for refactor then. Will try to do that.